zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterResultMessage.java

```diff
@@ -16,18 +16,30 @@
  */
 package org.apache.zeppelin.interpreter;
 
+import org.apache.zeppelin.tabledata.ColumnDef;
+
 import java.io.Serializable;
+import java.util.List;
 
 /**
  * Interpreter result message
  */
 public class InterpreterResultMessage implements Serializable {
   InterpreterResult.Type type;
   String data;
+  MessageColumnTypes msgColumnTypes;
 
   public InterpreterResultMessage(InterpreterResult.Type type, String data) {
     this.type = type;
     this.data = data;
+    this.msgColumnTypes = new MessageColumnTypes();
+  }
+
+  public InterpreterResultMessage(InterpreterResult.Type type, String data,
+                                  List<ColumnDef.TYPE> columnTypes) {
+    this.type = type;
+    this.data = data;
+    this.msgColumnTypes = new MessageColumnTypes(columnTypes);
```

Comment:
The below comments are related to this field. I think it's better to keep columnTypes itself.